src/main/java/com/yugabyte/sample/apps/CassandraInserts.java

```diff
@@ -0,0 +1,217 @@
+  }
+
+  private PreparedStatement getPreparedSelect() {
+    return getPreparedSelect(selectStatement, appConfig.localReads);
+  }
+
+  @Override
+  protected PreparedStatement getPreparedInsert() {
+    return getPreparedInsert(insertStatement);
+  }
+
+  private static BoundStatement getBoundInsertStatement(PreparedStatement ps, Key key) {
+    BoundStatement bs = ps.bind();
+    bs.setString(0, key.asString());
+    for (int i = 0; i < countValCols; ++i) {
+      bs.setString(1 + i, key.getValueStr(i, appConfig.valueSize));
+    }
+    return bs;
+  }
+
+  @Override
+  public long doRead() {
+    Key key = getSimpleLoadGenerator().getKeyToRead();
+    if (key == null) {
+      return 0;
+    }
+    BoundStatement bound = getPreparedSelect().bind(key.asString());
+    ResultSet rs = getCassandraClient().execute(bound);
+    List<Row> rows = rs.all();
+    LOG.debug("Read key: " + key.asString() + " return code" + rs.toString());
+    if (rows.size() != 1) {
+      LOG.fatal("Read key: " + key.asString() + " expected 1 row in result, got " + rows.size());
+    }
+    if (!key.asString().equals(rows.get(0).getString(0))) {
+      LOG.fatal("Read key: " + key.asString() + ", got " + rows.get(0).getString(0));
+    }
+    return 1;
+  }
+
+  @Override
+  public long doWrite(int threadIdx) {
+    HashSet<Key> keys = new HashSet<Key>();
+    try {
+      if (appConfig.batchWrite) {
+        BatchStatement batch = new BatchStatement();
+        PreparedStatement insert = getPreparedInsert();
+        for (int i = 0; i < appConfig.batchSize; i++) {
+          Key key = getSimpleLoadGenerator().getKeyToWrite();
+          keys.add(key);
+          batch.add(getBoundInsertStatement(insert, key));
+        }
+        ResultSet resultSet = getCassandraClient().execute(batch);
+        LOG.debug("Wrote keys count: " + keys.size() + ", return code: " + resultSet.toString());
+        for (Key key : keys) {
+          getSimpleLoadGenerator().recordWriteSuccess(key);
+        }
+        return keys.size();
+      } else {
+        Key key = getSimpleLoadGenerator().getKeyToWrite();
+        if (key == null) {
+          return 0;
+        }
+        keys.add(key);
+        ResultSet resultSet =
+            getCassandraClient().execute(getBoundInsertStatement(getPreparedInsert(), key));
+        LOG.debug("Wrote key: " + key.toString() + ", return code: " + resultSet.toString());
+        getSimpleLoadGenerator().recordWriteSuccess(key);
+        return 1;
+      }
+    } catch (Exception e) {
+      for (Key key : keys) {
+        getSimpleLoadGenerator().recordWriteFailure(key);
+      }
+      throw e;
+    }
+  }
+
+  @Override
+  public List<String> getWorkloadDescription() {
+    return Arrays.asList(
+        "Secondary index on key-value YCQL table. Writes unique keys with an index on values. ",
+        "Note that these are strongly consistent, global secondary indexes. Supports a flag to run at",
```

Comment:
Two contradictory sentences: "note that these are strongly consistent indexes", but then you're saying they can be at "app enforced consistency". Please re-word.

Reply:
Done.